### ISSUE TITLE
ESS blocks Emacs regardless of ess-eval-visibly 

### ISSUE THREAD

**Issue by jabranham** [emacs-ess/ESS#535]
I have `ess-eval-visibly` set to `'nowait` but sending code to the inferior process can still sometimes block Emacs. Hitting `C-g` three times always solves the problem (and the code continues to get evaluated). 

Has anyone else ran into this issue or have ideas on how to debug this?

**Comment by vspinu** [comment]
Yes. Happens in eldoc and completion system. I have a fix in mind. Now that we have an open issue for it will try to fix asap :)

**Comment by mmaechler** [comment]
Yes, I had encountered it many times too (and worked around with 3x C-g ) but hadn't been brave enough to raise a vague issue (as I usually hear: "does not happen for us; reproducible example please" ;-\).  Thanks for addressing it!

**Comment by rsparapa** [comment]
Yes, I see this too.  However, it seems to be sporadic.
